### PR TITLE
7.post articles id comments

### PR DESCRIPTION
**Pull request by GaryCale**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -121,31 +118,49 @@ describe("Endpoint: GET /api/articles/:article_id", () => {
...
-//         });
-//       });
-//   });
-// });
+describe("Endpoint: GET /api/articles/:article_id/comments", () => {
+  test("Returns a status code of 200 when query exists", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+  });
```
Comment (NRMSMITH) [inline __tests__/app.test.js]: You don't need to do this test as you're expecting 200 in the one below :D

File: __tests__/app.test.js [hunk 2]
```diff
@@ -121,31 +118,49 @@ describe("Endpoint: GET /api/articles/:article_id", () => {
...
+  test("Returns 400 when passed an invalid id", () => {
+    return request(app)
+      .get("/api/articles/abc/comments")
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad Request");
+      });
+  });
+});
```
Comment (NRMSMITH) [inline __tests__/app.test.js:202]: Add another test - I think you need to have a look at a 200 valid id but there are no comments :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -121,31 +119,96 @@ describe("Endpoint: GET /api/articles/:article_id", () => {
...
+  });
+  test("Returns 404 when passed a valid id but it does not exist", () => {
+    return request(app)
+      .get("/api/articles/999/comments")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Comments not found");
+      });
+  });
+});
```
Comment (NRMSMITH) [inline __tests__/app.test.js:265]: A few more tests to add: 

201 -> ignore unnecessary properties in object that is sent
404 -> username does not exist
400 -> missing required fields

File: __tests__/app.test.js [hunk 4]
```diff
@@ -121,31 +119,96 @@ describe("Endpoint: GET /api/articles/:article_id", () => {
...
+        expect(res.body.msg).toBe("Bad Request");
+      });
+  });
+  test("Returns 404 when passed a valid id but it does not exist", () => {
+    return request(app)
+      .get("/api/articles/999/comments")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Comments not found");
+      });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:227]: This test is confusing -> surely the message should be that the id cannot be found?

File: controllers/controllers.js [hunk 5]
```diff
@@ -36,19 +37,34 @@ const getSingleArticleById = (request, response, next) => {
...
+  fetchComments(article_id)
+  .then((comments) => {
+    response.status(200).send({ comments: comments });
+  })
+  .catch((err) => {
+    next(err)
+  })
+};
+
+const postArticle = (req, res, next) => {
```
Comment (NRMSMITH) [inline controllers/controllers.js:52]: we're posting comments not articles -> please make sure you change your function name

File: controllers/controllers.js [hunk 6]
```diff
@@ -36,19 +37,34 @@ const getSingleArticleById = (request, response, next) => {
...
+  console.log(req.body)
+  const { article_id } = req.params;
+  console.log(article_id)
+  insertComment(req.body, article_id)
+  .then((newArticle) => {
+    res.status(201).send({ article: newArticle });
```
Comment (NRMSMITH) [inline controllers/controllers.js:56]: should be sent back as a comment, not an article

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Some tests need to be updated/added & naming of your controller function, and what you're sending to the tests.

**Review by MKD01 — APPROVED** [review]
Happy for this to be merged

**Pull request merged**